model/registry/user.yaml

```diff
@@ -28,8 +28,17 @@ groups:
         type: string
         stability: experimental
         brief: >
-          Unique identifier of the user.
+          Unique identifier of an authenticated user.
         examples: ['S-1-5-21-202424912787-2692429404-2351956786-1000']
+      - id: anonymous_id
+        type: string
+        stability: experimental
+        brief: >
+          Unique identifier of an anonymous user who interacts with a system or service without providing personal information or authentication credentials. 
```

Comment:
Would anonymous_id and user_id be mutually exclusive or not for a single event?
For example, if a user started installing an app, once the app is installed, the user provided credential and logged in, do we expect the telemetry to have BOTH anonymous_id and user_id after the successful login?

Reply:
I think they cannot be mutually exclusive, otherwise you cannot calculate correct user counts like MAU for scenarios that need to count anonymous users. 

If mutually exclusive, `anonymous_id` will be missing users who are signed in with long term tokens (never became anonymous in the query period). On the other hand, trying to calculate MAU by `dcount(user_id)` + `dcount(anonymous_id)` will duplicate users who _did_ sign in during the query period.

Reply:
I agree with you @OsvaldoRosado, the current wording sounds like user_id is for non-anonymous, and anonymous_id is for anonymous.

Reply:
@OsvaldoRosado do I understand correctly that in your example we would have a link between anonymous and authenticated user? Because telemetry will provide both ids if available

Reply:
@heyams how can we distinguish between not authenticated and non-anonymous user? If the user hasn't authenticate itself, do we have means to know if he is anonymous (doesn't have an account) or not anonymous (have account but hasn't logged in)?

Reply:
I asked Copilot about the difference between authenticated user id, non-anonymous user id, and unauthenticated user id: 

```
Copilot
Certainly! Let’s break down the differences between authenticated user IDs, non-anonymous user IDs, and unauthenticated user IDs:

Authenticated User IDs:
Definition: Authenticated user IDs represent entities that have successfully proven their identity to the system. These entities are typically associated with user accounts or processes running in a specific user’s security context.
Unique Identifier: Each authenticated user has a unique Security Identifier (SID) issued by an authority (e.g., a Windows domain controller). The SID is stored in a security database.
Access Tokens: When a user logs in, the system creates an access token containing the user’s SID, user rights, and group SIDs. This token defines the security context for the user’s actions.
Examples: Regular user accounts, computer accounts, and other authenticated entities fall into this category.
Non-Anonymous User IDs:
Definition: Non-anonymous user IDs refer to known entities that haven’t undergone full authentication. They may have limited rights on the system.
Examples:
Guest Account: The Guest account, even if it has a password, remains non-anonymous because it lacks full authentication.
Built-in Accounts: Certain built-in accounts (e.g., LOCAL_SERVICE) are also non-anonymous.
Note: Non-anonymous users don’t need credentials to log in, but their identities are still known.
Unauthenticated User IDs:
Definition: Unauthenticated user IDs represent entities with no verified identity. These are often transient or anonymous.
Examples:
Anonymous Users: Visitors accessing public resources without logging in.
Temporary Tokens: Tokens issued for short-term access (e.g., OAuth tokens before authentication).
Note: Unauthenticated IDs lack a persistent association with a specific user.
In summary, authenticated user IDs are fully verified, non-anonymous user IDs are known but unauthenticated, and unauthenticated user IDs lack any verified identity. 
```

Reply:
> @OsvaldoRosado do I understand correctly that in your example we would have a link between anonymous and authenticated user? Because telemetry will provide both ids if available

Yes you would. "Anonymous" is arguably a confusing term here. I see it as basically a uniquely identifying user id that is consistently available for all users, regardless of authentication state.

Reply:
@trisch-me I've updated the description of `anonymous_id` for better clarity. I hope this will alleviate some of the confusion.